src/main/java/org/assertj/core/internal/Arrays.java

```diff
@@ -208,11 +208,22 @@ void assertDoesNotContain(AssertionInfo info, Failures failures, Object array, O
 
   void assertContainsOnly(AssertionInfo info, Failures failures, Object actual, Object values) {
     if (commonChecks(info, actual, values)) return;
-    IterableDiff diff = diff(asList(actual), asList(values), comparisonStrategy);
-    if (diff.differencesFound())
-      throw failures.failure(info, shouldContainOnly(actual, values,
-                                                     diff.missing, diff.unexpected,
-                                                     comparisonStrategy));
```

Comment:
I might miss a point but if `IterableDiff` is fixed, why do we need to change this method?